rust/protocol/src/sealed_sender.rs

```diff
@@ -594,96 +598,186 @@ impl UnidentifiedSenderMessage {
+        let recipient_identity = IdentityKeyPair::generate(&mut rand::thread_rng());
+
+        // Generate an ephemeral key pair.
+        let sender_ephemeral = KeyPair::generate(&mut rand::thread_rng());
+        let ephemeral_public = sender_ephemeral.public_key;
+        // Generate ephemeral cipher, chain, and MAC keys.
+        let sender_eph_keys = EphemeralKeys::calculate(
+            &sender_ephemeral,
+            recipient_identity.public_key(),
+            Direction::Sending,
+        )?;
 
-        pub fn mac_key(&self) -> Result<&[u8]> {
-            Ok(&self.derived_values[64..96])
-        }
+        // Encrypt the sender's public key with AES-256 CTR and a MAC.
+        let sender_static_key_ctext = crypto::aes256_ctr_hmacsha256_encrypt(
+            &sender_identity.public_key().serialize(),
+            sender_eph_keys.cipher_key.as_ref(),
+            sender_eph_keys.mac_key.as_ref(),
+        )?;
+
+        // Generate another cipher and MAC key.
+        let sender_static_keys = StaticKeys::calculate(
+            &sender_identity,
+            recipient_identity.public_key(),
+            &sender_eph_keys.chain_key,
+            &sender_static_key_ctext,
+        )?;
+
+        let sender_message_contents = b"this is a binary message";
+        let sender_message_data = crypto::aes256_ctr_hmacsha256_encrypt(
+            sender_message_contents,
+            sender_static_keys.cipher_key.as_ref(),
+            sender_static_keys.mac_key.as_ref(),
+        )?;
+
+        // The message recipient calculates the ephemeral key and the sender's public key.
+        let recipient_eph_keys = EphemeralKeys::calculate(
+            &recipient_identity.into(),
+            &ephemeral_public,
+            Direction::Receiving,
+        )?;
+        assert_eq!(sender_eph_keys, recipient_eph_keys);
+
+        let recipient_message_key_bytes = crypto::aes256_ctr_hmacsha256_decrypt(
+            &sender_static_key_ctext,
+            recipient_eph_keys.cipher_key.as_ref(),
+            recipient_eph_keys.mac_key.as_ref(),
+        )?;
+        let sender_public_key = PublicKey::try_from(recipient_message_key_bytes.as_ref())?;
+        assert_eq!(sender_identity.public_key(), &sender_public_key);
+
+        let recipient_static_keys = StaticKeys::calculate(
+            &recipient_identity,
+            &sender_public_key,
+            &recipient_eph_keys.chain_key,
+            &sender_static_key_ctext,
+        )?;
+
+        let recipient_message_contents = crypto::aes256_ctr_hmacsha256_decrypt(
+            &sender_message_data,
+            recipient_static_keys.cipher_key.as_ref(),
+            recipient_static_keys.mac_key.as_ref(),
+        )?;
+        assert_eq!(recipient_message_contents, sender_message_contents);
+
+        Ok(())
     }
 }
 
+/// Encrypt the plaintext message `ptext`, generate an [UnidentifiedSenderMessageContent], then pass
+/// the result to [sealed_sender_encrypt_from_usmc].
```

Comment:
It's pretty clear that `sealed_sender_encrypt_from_usmc` is the place to go from the summary, but I think it could still stand a body line like "This is a simple way to encrypt a message in a 1:1 session using [Sealed Sender v1]." It's just a shortcut, albeit one that made sense as presented API before SenderKey came around.

Reply:
Thanks for this clarification! I've made this change! I will probably try to parse/document the `SenderKey` mechanism next :)))